api-lib/sendSocialMessage.ts

```diff
@@ -176,16 +216,58 @@ export async function sendSocialMessage({
       chat_id: channelId,
       text: msg,
     };
-    const res = await fetch(`${TELEGRAM_BOT_BASE_URL}/sendMessage`, {
+
+    const updateTelegramBot = fetch(`${TELEGRAM_BOT_BASE_URL}/sendMessage`, {
       method: 'POST',
       body: JSON.stringify(telegramBotPost),
       headers: {
         'Content-Type': 'application/json',
       },
-    });
+    })
+      .catch(err => {
+        console.error(
+          'Error updating telegram-bot:',
+          err.message,
+          `, Circle Id: ${circle?.id}`,
+          `, Channel Id: ${channelId}`
+        );
+        throw new Error(err);
+      })
+      .then(res => {
+        if (!res.ok) {
+          console.error(
+            'Error updating telegram-bot:',
+            res.status,
+            res.statusText,
+            `, Channel Id: ${channelId}`,
+            `, Circle Id: ${circle?.id}`,
+            `, notifyOrg: ${notifyOrg}`
+          );
+          return res.json().then(json => {
+            throw new Error(JSON.stringify(json));
+          });
+        }
+      });
+    requests.push(updateTelegramBot);
+  }
+
+  const responses = await Promise.allSettled(requests);
 
-    if (!res.ok) {
-      throw new Error(JSON.stringify(await res.json()));
-    }
+  const errors = responses.filter(isRejected);
+
+  if (errors.length > 0) {
```

Comment:
now, what to do if there are some errors and some successes is an interesting question. i think we should probably not have the whole thing throw an error if we have at least one success.

my thinking here is that having at least one success means at least one message got through. if we were to repeat this job due to having one error, that message would be duplicated.

the tradeoff is, if one of the errors was due to a transient condition, we would miss the opportunity to retry it and have it work the second time.

so there's a product choice here between "too few messages--but at least one success", and "too many messages".

Reply:
after discussion, i think this is probably fine as-is. if we do send duplicate messages to a community in one channel because of errors with another channel, that will at least help raise awareness that there's a problem